LogLeadRecords.cls

```diff
@@ -0,0 +1,35 @@
+/*
+Create a custom object called as 'Logs'. Create a long text area field called as 'Error'. Create 100 
+Lead records using DML operations having a unique name. For all records which were not inserted into 
+Lead object, insert a record in Log object along with the reason why a record was not inserted. Also, 
+execute assignment rules to auto allocate Lead records to correct owner. 

```

Comment:
This is missing here. You need to execute assignment rule from apex.

- [ ] Execute assignment rule from apex